components/merkledb/src/proof_map_index/key.rs

```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// This lint is triggered
```

Comment:
I'd add an explanation from @slowli [comment](https://github.com/exonum/exonum/pull/1558/files/430f5ff361563109d703d7778a7777c35e4aa6ec..a563a313f4d7520a2b60c3ce047edef007500133#r349547834) for future readers.